airbyte-cdk/java/airbyte-cdk/db-sources/src/main/kotlin/io/airbyte/cdk/components/debezium/StateFilesAccessor.kt

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+package io.airbyte.cdk.components.debezium
+
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.google.common.collect.ImmutableList
+import io.airbyte.commons.jackson.MoreMappers
+import io.debezium.config.Configuration
+import io.debezium.relational.history.AbstractFileBasedSchemaHistory
+import io.debezium.relational.history.HistoryRecord
+import io.debezium.relational.history.HistoryRecordComparator
+import io.debezium.relational.history.SchemaHistoryListener
+import io.debezium.storage.file.history.FileSchemaHistory
+import java.io.IOException
+import java.io.UncheckedIOException
+import java.lang.reflect.InvocationTargetException
+import java.lang.reflect.Method
+import java.nio.ByteBuffer
+import java.nio.file.Files
+import java.nio.file.Path
+import java.util.function.Consumer
+import kotlin.Exception
+import kotlin.RuntimeException
+import kotlin.String
+import org.apache.commons.io.FileUtils
+import org.apache.kafka.common.utils.ByteBufferInputStream
+import org.apache.kafka.connect.json.JsonConverter
+import org.apache.kafka.connect.json.JsonConverterConfig
+import org.apache.kafka.connect.runtime.WorkerConfig
+import org.apache.kafka.connect.runtime.standalone.StandaloneConfig
+import org.apache.kafka.connect.storage.Converter
+import org.apache.kafka.connect.storage.FileOffsetBackingStore
+
+/**
+ * [StateFilesAccessor] reads and writes from Debezium offset and schema history files. This should
+ * NOT be done while the Debezium Engine is running! None of this is thread-safe. Instead, this
```

Comment:
There is a case we tinker with the files while this is running (when acking debezium lsn as we go along syncing perhaps)
I think in today's code there's a chance we bump into debezium trying to write the file concurrently

[uploading Screenshot 2024-03-08 at 11.28.20 AM.png…]
![Screenshot 2024-03-08 at 11.28.20 AM.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/otXb76ykserkfI6nLz16/b2c5d233-242f-400d-a518-e13f79257496.png)